grid2op/Opponent/OpponentSpace.py

```diff
@@ -107,14 +119,35 @@ def attack(self, observation, agent_action, env_action):
             The attack the opponent wants to perform (or "do nothing" if the attack was too costly)
 
         """
+
+        # Update variables
         self.budget += self.budget_per_timestep
-        attack = self.opponent.attack(observation, agent_action, env_action, self.budget, self.previous_fails)
-        tmp = self.compute_budget(attack)
-        if tmp <= self.budget:
-            self.previous_fails = False
-            self.budget -= tmp
-            res = attack
+        self.current_attack_duration = max(0, self.current_attack_duration - 1)
+        self.current_attack_cooldown = max(0, self.current_attack_cooldown - 1)
+
+        # If currently attacking
+        if self.current_attack_duration > 0:
+            attack = self.current_attack
+
+        # If the opponent has already attacked today
+        elif self.current_attack_cooldown > self.attack_cooldown:
+            attack = self._do_nothing
+
+        # If the opponent can attack  
         else:
-            self.previous_fails = True
-            res = self._do_nothing
-        return res
+            self.previous_fails = False
+            attack = self.opponent.attack(observation, agent_action, env_action, self.budget,
```

Comment:
the opponent is forced to do the same attack in this setting no ? So why ask him what he wants to do ?
Add a method in opponent like `tell_attack_continue()` that does nothing, and just return `self.last_attack, self.current_attack_duration`

Reply:
No, the git diffs may not show it because I rewrote the function but this is actually the only part where the opponent has to choose what to do

Reply:
It should be good to have a method like `tell_attack_continues(observation, agent_action, env_action, self.budget)` that is called each time steps for example.
This would allow opponent to keep track of what's going on when their attack is being performed for example, we might need that for more "advanced" opponents than just random.

Reply:
done